### PR TITLE
HDDS-1848. Fix TestOzoneManagerHA and TestOzoneManagerSnapShotProvider.

### PR DESCRIPTION
**Pull request by bharatviswa504**

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/MiniOzoneHAClusterImpl.java [hunk 1]
```diff
@@ -169,6 +169,11 @@ public MiniOzoneCluster build() throws IOException {
         throw new IllegalArgumentException("Number of active OMs cannot be " +
             "more than the total number of OMs");
       }
+
+      // If num of ActiveOMs is not set, set it to numOfOMs.
```
Comment (arp7) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/MiniOzoneHAClusterImpl.java]: Should this be set before the check on line 168?

Also why not set it always. Would that be wrong?

Comment (bharatviswa504) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/MiniOzoneHAClusterImpl.java]: If numOfActiveOMs is not set we set with numofOMs (which is total count of OMS). When user set some value to numOfActiveOMs we don't want to override that, this is the reason for doing that.

If it is not set, we don't need to do this check
if (numOfActiveOMs > numOfOMs) {
        throw new IllegalArgumentException("Number of active OMs cannot be " +
            "more than the total number of OMs");
      }
So, it is okay to add after this too.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 534 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 582 | trunk passed |
| +1 | compile | 344 | trunk passed |
| +1 | checkstyle | 59 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 780 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 142 | trunk passed |
| 0 | spotbugs | 410 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 594 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 530 | the patch passed |
| +1 | compile | 348 | the patch passed |
| +1 | javac | 348 | the patch passed |
| +1 | checkstyle | 62 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 616 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 152 | the patch passed |
| +1 | findbugs | 673 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 282 | hadoop-hdds in the patch passed. |
| -1 | unit | 1768 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 70 | The patch does not generate ASF License warnings. |
| | | 7697 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.container.common.statemachine.commandhandler.TestBlockDeletion |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestReadRetries |
|   | hadoop.ozone.client.rpc.TestFailureHandlingByClient |
|   | hadoop.ozone.TestStorageContainerManager |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1148/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1148 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux f6cd43ecf9da 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / ecb9f81 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1148/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1148/1/testReport/ |
| Max. process+thread count | 4433 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/integration-test U: hadoop-ozone/integration-test |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1148/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by bharatviswa504** [comment]
Tests are passing now.
Other test failures are not related to this patch. I will commit this to the trunk.

**Pull request merged**